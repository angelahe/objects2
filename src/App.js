import React, { Component } from 'react';
import './App.css';
//import Starter from './components/Starter';
import MathComp from './components/MathComp';
import AccountUI from './components/AccountUI';
import AccountsUI from './components/AccountsUI';
import CommunityUI from './components/CommunityUI';
import LinkedListUI from './components/LinkedListUI';

import NavBar from './components/NavBar';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            navClicked: "default",
            message: "Starting message"
        };
    }

    onNavClickChanged = (clickBtn) => {
//        console.log("in onNavClickChanged, clickbtn is", clickBtn);
        this.setState({navClicked : clickBtn, message : clickBtn + " was clicked"});
    }

    render() {

        return (

            <div className="App">
                <NavBar navClicked={this.onNavClickChanged}/>
                {(this.state.navClicked==="linked") ? <LinkedListUI /> : null}
                {(this.state.navClicked==="math") ? <MathComp /> : null}
                {(this.state.navClicked==="account") ? <AccountUI /> : null}
                {(this.state.navClicked==="accounts") ? <AccountsUI /> : null}
                {(this.state.navClicked==="community") ? <CommunityUI/> : null}
            </div>
        );
    }
}

export default App;
