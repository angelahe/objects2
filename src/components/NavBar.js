import React from "react";
import '../App.css';

import analytics from '../images/analytics.svg';
import creative from '../images/creative.svg';
import leader from '../images/leader.svg';
import teamwork from '../images/teamwork.svg';
import teamwork2 from '../images/teamwork2.svg';

const NavBar = (props) => {

    function handleClick(e) {
        console.log("in handleClick function");
        console.log(props.navClicked);
        props.navClicked(e.target.getAttribute("navbtn"));
    }

    return (
        <div>
            <header>
                <img src={analytics} navbtn="default" onClick={handleClick} className="App-icon" alt="one"/>
                <img src={creative} navbtn="math" onClick={handleClick} className="App-icon" alt="two"/>
                <img src={leader} navbtn="leader" onClick={handleClick} className="App-icon" alt="three"/>
                <img src={teamwork} navbtn="teamwork" onClick={handleClick} className="App-icon" alt="four"/>
                <img src={teamwork2} navbtn="teamwork2" onClick={handleClick} className="App-icon" alt="five"/>
            </header>
            <p>{props.message}</p>
        </div>
    );
}
export default NavBar
