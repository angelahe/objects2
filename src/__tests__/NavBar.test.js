import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NavBar from "../components/NavBar";
import App from "../App";
import renderer from "react-test-renderer";

const props = new App();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBar navClicked={props.onNavClickChanged}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<NavBar navClicked={props.onNavClickChanged}/>);
});

test('MathComp renders correctly', () => {
  const tree = renderer.create(<NavBar navClicked={props.onNavClickChanged}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

xtest('call navClicked with correct argument when click default app', () => {
  const didClick = jest.fn();
  const wrapper = shallow(
    <NavBar navClicked={props.onNavClickChanged}/>
  );
  const n = wrapper.find({navbtn: 'default'});
  const mockedEvent = {target: {navbtn: "default"}};

  //const mockedEvent = {e: {}}
  //n.simulate('click', mockedEvent);
  //n.simulate('click', {e.target.getattribute: "navbtn") = "default"}});
  n.simulate('click', mockedEvent);
  expect(navClicked).toBeCalledWith("default");
});