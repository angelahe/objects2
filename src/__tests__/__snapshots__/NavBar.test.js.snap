// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathComp renders correctly 1`] = `
<div>
  <header>
    <img
      alt="one"
      className="App-icon"
      navbtn="default"
      onClick={[Function]}
      src="analytics.svg"
    />
    <img
      alt="two"
      className="App-icon"
      navbtn="math"
      onClick={[Function]}
      src="creative.svg"
    />
    <img
      alt="three"
      className="App-icon"
      navbtn="account"
      onClick={[Function]}
      src="leader.svg"
    />
    <img
      alt="four"
      className="App-icon"
      navbtn="accounts"
      onClick={[Function]}
      src="teamwork.svg"
    />
    <img
      alt="five"
      className="App-icon"
      navbtn="community"
      onClick={[Function]}
      src="teamwork2.svg"
    />
  </header>
  <p />
</div>
`;
